drivers/adc/Kconfig.stm32f4

```diff
@@ -0,0 +1,373 @@
+# Kconfig - ADC configuration options
+
+#
+# Copyright (c) 2015 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig ADC_STM32F4
+	bool "STM32F4 ADC driver"
+	depends on SOC_FAMILY_STM32
+	select USE_STM32_HAL_GPIO
+	select USE_STM32_HAL_GPIO_EX
+	select USE_STM32_HAL_ADC
+	select USE_STM32_HAL_ADC_EX
+	select USE_STM32_HAL_CORTEX
+	help
+		Enable the driver implementation for the stm32f4xx ADC
+
+if ADC_STM32F4
+
+	menu ADC_0
```

Comment:
Don't indent here. Kconfig files should remain flat

Reply:
think the best solution would be a flat ADC driver + the instance setup (like channels) in the dts. Never done this before. Is there a document/example that describes how to connect between dts node and driver?

Reply:
@StefJar , there is a doc there: http://docs.zephyrproject.org/devices/dts/device_tree.html

To sum up:
1- Populate the node in device tree (you can base initially on linux dts)
2- Provide a yaml file (here: `dts/bindings/iio/adc/`) to instruct dts script on how to interpret the adc node. File name should be same as node compatible. With this, at build, you should get a bunch of #define's generated in output folder, in `zephyr/include/generated/generated_dts_board.h`.
3-Provide CONFIG_XXX to generated #define's matching in .fixup files (`arch/arm/soc/st_stm32/stm32f4/dts.fixup` for  series)
4- Define HAS_DTS_ADC for the series where adc device node is populated

Then you're good to use whatever values populated in device tree directly in your driver

Reply:
> Then you're good to use whatever values populated in device tree directly in your driver

Note that the device tree should describe the hardware, not the configuration.

It would be better to check the Linux driver (https://elixir.bootlin.com/linux/v4.17/source/drivers/iio/adc/stm32-adc-core.c and https://elixir.bootlin.com/linux/v4.17/source/drivers/iio/adc/stm32-adc.c) and corresponding DTS binding (https://elixir.bootlin.com/linux/v4.17/source/Documentation/devicetree/bindings/iio/adc/st,stm32-adc.txt) to make sure that we stay compatible.

Reply:
merci - I will take a deeper look when I have time for that